src/es-module-shims.js

```diff
@@ -267,15 +267,23 @@ function resolveDeps (load, seen) {
         if (cycleShell) {
           // circular shell creation
           if (!(blobUrl = depLoad.s)) {
+            function dotOrBracketAccessor(expt) {
+              const ch = depLoad.S[expt.s];
+              return (ch === '"' || ch === "'") ? `[${depLoad.S.slice(expt.s, expt.e)}]` : '.' + expt.n;
+            }
             blobUrl = depLoad.s = createBlob(`export function u$_(m){${
               depLoad.a[1].map(
-                name => name === 'default' ? `d$_=m.default` : `${name}=m.${name}`
+                expt => expt.ln ? `${expt.ln}=m${dotOrBracketAccessor(expt)}` : `d$_=m.default`
               ).join(',')
             }}${
-              depLoad.a[1].map(name =>
-                name === 'default' ? `let d$_;export{d$_ as default}` : `export let ${name}`
-              ).join(';')
-            }\n//# sourceURL=${depLoad.r}?cycle`);
+              depLoad.a[1].map(
+                expt => expt.ln ? `let ${expt.ln};` : 'let d$_;'
+              ).join('')
+            }export {${
+              depLoad.a[1].map(
+                expt => expt.ln ? `${expt.ln} as ${depLoad.S.slice(expt.s, expt.e)}` : `d$_ as default`
```

Comment:
Default should work out naturally, we shouldn't need to branch on it anymore?

Reply:
I don't understand how. We still need to export `default`, but it doesn't have an `ln`, right?

Moved the `expr.ln` conditional to a `map()` ahead of the template expansion.

Reply:
This is just an _early_ export process to help with during-execution accesses - the main cycle update function is called after execution with the full namespace, which would have the default and reexported bindings as well.